### PR TITLE
refactor: `direction` and `qOverP` for Fatras particle interface

### PR DESCRIPTION
**Pull request by andiwand**
I noticed that we are quite inconsistent with the use of `direction` and `unitDirection`. Personally I would opt for `direction` and have it as a unit vector implicitly.

Also, I changed `qop` to `qOverP` which I will also align for the Stepper and TrackParameter interfaces in follow-up PRs

I pulled these changes out of https://github.com/acts-project/acts/pull/2181.

### CODE REVIEW

**Comment by paulgessinger** [comment]
Still has some compile failures. Marking as WIP until they are fixed.

File: Fatras/include/ActsFatras/EventData/Particle.hpp [hunk 1]
```diff
@@ -154,11 +154,15 @@ class Particle {
...
   /// Particle mass.
   constexpr Scalar mass() const { return m_mass; }
-  /// Particl qop.
-  constexpr Scalar qop() const {
-    return (charge() == 0 ? 1 : charge()) / absoluteMomentum();
+
+  /// Particl qOverP.
+  constexpr Scalar qOverP() const {
+    return charge() == 0 ? 1 / absoluteMomentum()
+                         : charge() / absoluteMomentum();
```
Comment (AJPfleger) [inline Fatras/include/ActsFatras/EventData/Particle.hpp:165]: As far as I understand is this change not necessary due to the `unitDirection`->`direction` unification, or am I missing something?

Comment (andiwand) [reply Fatras/include/ActsFatras/EventData/Particle.hpp:165]: right but I would also like to change that so I will adopt the title and description

File: CI/physmon/fpe_masks.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 "Fatras/include/ActsFatras/Physics/ElectroMagnetic/BetheHeitler.hpp:66":
   FLTUND: 1
-"Fatras/include/ActsFatras/Kernel/detail/SimulationActor.hpp:178":
+"Fatras/include/ActsFatras/Kernel/detail/SimulationActor.hpp:177":
+  FLTUND: 1
+"Fatras/include/ActsFatras/Kernel/detail/SimulationActor.hpp:249":
```
Comment (andiwand) [inline CI/physmon/fpe_masks.yml:5]:
```suggestion
```

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/30890481) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/5861556)!

**Comment by paulgessinger** [comment]
This actually breaks compilation and requires a fix on the Athena side.

**Comment by andiwand** [comment]
Interesting - I did not know that we use Fatras in Athena

**Comment by paulgessinger** [comment]
Not in production, but it's being tested: https://gitlab.cern.ch/atlas/athena/-/tree/main/Simulation/ISF/ISF_Acts

**Comment by paulgessinger** [comment]
Fix is here: https://gitlab.cern.ch/atlas/athena/-/commit/9fa25fdf2feeab609e9566052abe664b3eca4b1f?merge_request_iid=63853

**Pull request merged**